devices/Mcp7940xx/tests/Mcp7940mTests.cs

```diff
@@ -0,0 +1,1354 @@
+using Iot.Device.Common;
+using Iot.Device.Mcp7940xx;
+using nanoFramework.Hardware.Esp32;
+using nanoFramework.TestFramework;
+using System;
+using System.Device.I2c;
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading;
+
+namespace Iot.Device.NFUnitTest
+{
+    [TestClass]
+    public class Mcp7940mTests
+    {
+        private static I2cDevice _i2cDevice;
+        private static Mcp7940m _clock;
+
```

Comment:
you should use this pattern for the test: https://github.com/nanoframework/System.IO.Ports/blob/c0c065f28ae5589d19436d725fad5422c9113316/Tests/UnitTestsSerialPort/SerialTests.cs#L23
Making sure you are on a hardware that supports the test. If not, just skip the Setup test and all tests will be skipped. This does allow the test to run on pipelines without having to run on a real hardware.